### PR TITLE
Add Kyverno admission latency alerts

### PR DESCRIPTION
**Pull request by bastjan**
- `KyvernoAverageAdmissionLatencyHigh`: Alerts if average admission latency reaches 5 seconds.
- `KyvernoAdmissionLatencyReachedWebhookTimeout`: Alerts if there are requests taking longer than the webhook timeout.

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: class/defaults.yml [hunk 1]
```diff
@@ -18,6 +18,41 @@ parameters:
...
+              message: Kyverno admission latency higher than Kubernetes webhook timeout (10s).
+              runbook_url: https://hub.syn.tools/kyverno/runbooks/KyvernoAdmissionLatencyReachedWebhookTimeout.html
+              summary: Kyverno admission too slow.
+            expr: |
+              (
+                sum by(resource_kind, resource_namespace) (increase(kyverno_admission_review_duration_seconds_bucket{le="+Inf"}[5m]))
+              -
+                sum by(resource_kind, resource_namespace) (increase(kyverno_admission_review_duration_seconds_bucket{le="10"}[5m]))
+              ) > 0
+            for: 5m
```
Comment (glrf) [inline class/defaults.yml:53]: I have two questions for this rule:

1. Does kyverno actually report admission latency over 10s or does it also have a timeout there? 
2. Couldn't this rule potentially miss a single slow admission? We take the increase over 5m. That means for a single slow admission this expression will be true for exactly 5m, right? So with the `for: 5m` we might miss this one or at the very least the alert will be immediately closed, or am I completely missing how these rules work? :sweat_smile:

Comment (bastjan) [reply class/defaults.yml:53]: > 1. Does kyverno actually report admission latency over 10s or does it also have a timeout there?

It does from my testing.

> 2. I think you are right. This is racy at worst and does nothing at best.

File: class/defaults.yml [hunk 2]
```diff
@@ -18,6 +18,41 @@ parameters:
...
+          enabled: true
+          rule:
+            annotations:
+              description: Kyverno admission latency for {{$labels.resouce_kind}}/{{$labels.resource_namespace}} higher than Kubernetes webhook timeout (10s).
+              message: Kyverno admission latency higher than Kubernetes webhook timeout (10s).
+              runbook_url: https://hub.syn.tools/kyverno/runbooks/KyvernoAdmissionLatencyReachedWebhookTimeout.html
+              summary: Kyverno admission too slow.
+            expr: |
+              (
+                sum by(resource_kind, resource_namespace) (increase(kyverno_admission_review_duration_seconds_bucket{le="+Inf"}[10m]))
```
Comment (glrf) [inline class/defaults.yml:49]: I would probably increase this range even further so that the alert is firing a bit longer.

Comment (bastjan) [reply class/defaults.yml:49]: Would you be happy with `30m` or `1h`?

Comment (glrf) [reply class/defaults.yml:49]: I think both would be fine. I just think as an on-call engineer if I get such an alert for 5 min I might just miss / ignore it :sweat_smile:

File: docs/modules/ROOT/pages/runbooks/KyvernoAdmissionLatencyReachedWebhookTimeout.adoc [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+== icon:glasses[] Overview
+
+Kyverno average admission latency is higher than the webhook timeout.
+Creating or updating Kubernetes object failed because of the admission latency.
+
+== icon:bug[] Steps for debugging
+
+* Check Kyverno memory and CPU usage / limits.
+* Check Kyverno logs
+* VSHN: Notify Tarazed
```
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/KyvernoAdmissionLatencyReachedWebhookTimeout.adoc:14]: I'm missing steps for addressing the cause of the alert: does it help to increase memory and CPU limits? What are common error messages and what can I do to fix them?

**Review by simu — CHANGES_REQUESTED** [review]
Implementation LGTM (except for the code duplication/dead code), but the runbooks are not very helpful unless you already know exactly how to operate Kyverno.

File: component/main.jsonnet [hunk 4]
```diff
@@ -1,4 +1,5 @@
 // main template for kyverno
+local alerts = import 'alerts.libsonnet';
```
[the remark this answers is not in the document]
Comment (bastjan) [reply component/main.jsonnet:2]: Yes I meant to delete it after moving it for monitoring

File: docs/modules/ROOT/pages/runbooks/KyvernoAdmissionLatencyReachedWebhookTimeout.adoc (hunk 3 above)
Comment (bastjan) [reply docs/modules/ROOT/pages/runbooks/KyvernoAdmissionLatencyReachedWebhookTimeout.adoc:14]: We don't really know. I guess I should add the client side throttling log message. It's the only one we know.

File: docs/modules/ROOT/pages/runbooks/KyvernoAverageAdmissionLatencyHigh.adoc [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+== icon:glasses[] Overview
+
+Kyverno average admission latency reached 5 seconds.
+Kubernetes object creation und updating is delayed.
+
+== icon:bug[] Steps for debugging
+
+* Check Kyverno memory and CPU usage / limits.
+* Check Kyverno logs
+* VSHN: Notify Tarazed
```
[the remark this answers is not in the document]
Comment (bastjan) [reply docs/modules/ROOT/pages/runbooks/KyvernoAverageAdmissionLatencyHigh.adoc:14]: We don't really know. I guess I should add the client side throttling log message. It's the only one we know.

File: docs/modules/ROOT/pages/runbooks/KyvernoAdmissionLatencyReachedWebhookTimeout.adoc (hunk 3 above)
Comment (bastjan) [reply docs/modules/ROOT/pages/runbooks/KyvernoAdmissionLatencyReachedWebhookTimeout.adoc:22]: I added a note about the client side throttling.

**Pull request merged**